cfg/githubfieldmapper.go

```diff
@@ -0,0 +1,235 @@
+package cfg
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/andygrunwald/go-jira"
+	"github.com/coreos/issue-sync/lib/models"
+	"strings"
+	"time"
+)
+
+type FieldMapper interface {
+	MapFields(issue *models.ExtendedGithubIssue) jira.IssueFields
+
+	GetFieldValue(jIssue *jira.Issue, fieldKey FieldKey) (interface{}, error)
+
+	// getFieldIDs requests the metadata of every issue field in the JIRA
+	// project, and saves the IDs of the custom fields used by issue-sync.
+	GetFieldIDs(client jira.Client) (map[FieldKey]string, error)
+}
+
+type DefaultFieldMapper struct {
+	Config *Config
+}
+
+type JsonFieldMapper struct {
+	Config *Config
+}
+
+func (m DefaultFieldMapper) GetFieldValue(jIssue *jira.Issue, fieldKey FieldKey) (interface{}, error) {
+	switch fieldKey {
+	case GitHubID:
+		fallthrough
+	case GitHubNumber:
+		return jIssue.Fields.Unknowns.Int(m.Config.GetCompleteFieldKey(fieldKey))
```

Comment:
Here's a case where I would extract a local to avoid duplicated code (jIssue.Fields.Unknowns).

Reply:
Agreed. I'll factor this out.